src/components/pools/actions/Invest/nft/NftEligibility.tsx

```diff
@@ -0,0 +1,100 @@
+import { useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { debounce } from 'lodash'
+
+import { genericSuspense } from '@/src/components/helpers/SafeSuspense'
+import { Search } from '@/src/components/pureStyledComponents/form/Search'
+import { DEBOUNCED_INPUT_TIME } from '@/src/constants/misc'
+import useNftEligible from '@/src/hooks/aelin/nft/useNftEligible'
+import { ParsedNftCollectionRules } from '@/src/utils/aelinPoolUtils'
+
+const Title = styled.h1`
+  color: ${({ theme: { colors } }) => colors.textColor};
+  font-weight: 600;
+  font-size: 1.4rem;
+  line-height: 1.4;
+`
+const SubTitle = styled.div`
+  color: ${({ theme: { colors } }) => colors.textColor};
+  font-weight: 400;
+  font-size: 1.25rem;
+  line-height: 1rem;
+  padding-bottom: 1rem;
+`
+
+const Eligible = styled.div`
+  color: ${({ theme }) => theme.colors.textColor};
+  font-size: 1.25rem;
+  font-weight: 400;
+  line-height: 1.2;
+  margin: 1rem;
+  min-height: 15px;
+`
+
+const EligibleError = styled.span`
+  color: #ff7777;
+`
+
+const Verifying = styled(Eligible)`
+  font-style: italic;
+`
+
+type NftEligibilityProps = {
+  rules: ParsedNftCollectionRules
+}
+
+type EligibilityProps = {
+  rules: ParsedNftCollectionRules
+  tokenId?: string
+}
+
+const Eligibility: React.FC<EligibilityProps> = genericSuspense(
+  ({ rules, tokenId }) => {
+    const eligible = useNftEligible(rules, tokenId)
+    return (
+      <Eligible>
+        {tokenId && tokenId !== '' ? (
```

Comment:
Use this pattern `{ tokenId && tokenId !== '' && (<Component />)` to avoid the ternary operator and the fragment.